### ISSUE TITLE
[Feature request] Improve Style Configurator "Language" and "Style" listboxes

### ISSUE THREAD

**Issue by pryrt** [notepad-plus-plus/notepad-plus-plus#15939]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the Issue

As currently implemented, the **Style** ListBox in the **Style Configurator** is too narrow to show the whole name of the style.   For example, without individually clicking, you cannot tell which **Style** is `Change History revert modified` vs `Change History revert origin`, because both show up as just `Change History rev`. 

![image](https://github.com/user-attachments/assets/4ede2667-408d-4e42-a35f-55ea9df285c2)

Many of the per-language styles also extend beyond the width of the ListBox.  And even the **Language** field is a bit too narrow; see, for example, `Tektronix extended HEX` getting cut off to `Tektronix ext`.

The GUI thus requires extra clicking to figure out which **Style** is which, which slows down finding the Style you're actually looking for.

It would be nice if there were some easier way to see the whole **Style** name, other than clicking on them individually until you've found the **Style** you are looking for.

### Describe the solution you'd like.

There are a variety of options that I and others have thought of (see the [Forum discussion](https://community.notepad-plus-plus.org/topic/26439/list-box-hover-text-for-style-configurator/)).  

I will list them from the least appealing to me to the one that I would prefer -- with some notes on each:

1. Resize the **Style** ListBox without changing the size of the **Style Configurator** dialog
   - this would take space away from the others, and wouldn't be very workable
2. Permanently resize the **Style Configurator** dialog to make it wider
   - the **Style Configurator** dialog is already pretty wide, and making it permanently wider could inconvenience other users
3. Make the **Style Configurator** dialog user-resizable, and stretch the ListBoxes to fill the extra space
   - the extra space could be split, so about 40% of the extra space goes to the **Language** ListBox, and about 60% of the extra space goes to the **Style** ListBox
   - there should be a "smallest-width-allowed" limit, so the dialog is never made any narrower than it currently is
   - being resizable makes the decision up to the _user_ rather than forcing the width upon them
4. Make the individual entries in the ListBox have Hover Text
   - ![image](https://github.com/user-attachments/assets/c56cde02-d4da-4930-bdf1-dbd8cf3ffa2b)
   - When the mouse cursor is hovering over a **Language** name or a **Style** name, the full name could be hovered over
   - In the [forum](https://community.notepad-plus-plus.org/topic/26439/list-box-hover-text-for-style-configurator), Ekopalypse suggested [LB_ITEMFROMPOINT](https://learn.microsoft.com/en-us/windows/win32/controls/lb-itemfrompoint)
   - I found this SO question about [different tooltips for each item in ListBox](https://stackoverflow.com/questions/192584/how-can-i-set-different-tooltip-text-for-each-item-in-a-listbox)
5. Changing the **Language** ListBox to a ComboBox, which would make both of the **Language** and the **Style** lists wider. (This was [suggested by mpheath](https://community.notepad-plus-plus.org/topic/26439/list-box-hover-text-for-style-configurator) in the same Forum discussion, and immediately became my favorite idea) -- here's mpheath's mockup, showing the old arrangement vs the new:
   - ![image](https://github.com/user-attachments/assets/5ce34dd2-1cbb-4f64-acac-c209e2dcea62)
   - This solution looks really clean to me, and improves both the **Language** and the **Style** lists in a very straightforward fix

To sum up, my preference would be suggestion 5 (mpheath's suggestion of the **Language** ComboBox over the **Style** ListBox).  But any of suggestions 3 - 5 would make it easier to read the **Language** and **Style** names while using the **Style Configurator**.

### Debug Information

```shell
Notepad++ v8.7.4   (64-bit)
Build time : Dec  4 2024 - 23:50:05
Path : C:\usr\local\apps\npp\npp.8.7.4.portable.x64\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : ON
Cloud Config : OFF
Periodic Backup : ON
Placeholders : OFF
DirectWrite : ON
Multi-instance Mode : monoInst
File Status Auto-Detection : cdEnabledNew (for current file/tab only)
Dark Mode : OFF
OS Name : Windows 10 Enterprise (64-bit)
OS Version : 22H2
OS Build : 19045.5131
Current ANSI codepage : 1252
Plugins : none
```

**Comment by alankilborn** [comment]
> screenshot in point 5

Of course, the suggestion is not really to move _Global override_ back to the top (contrast with 8.7.4. where it is at the bottom).
:-)

**Comment by pryrt** [comment]
> the suggestion is not really to move Global override back to the top

Definitely not. I just didn't want to have to re-create mpheath's image without showing Global Override in the old location. ;-)
